core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -111,6 +112,10 @@ public final class CoreSocketFactory {
   private static CoreSocketFactory coreSocketFactory;
   private final ListenableFuture<KeyPair> localKeyPair;
   private final ConcurrentHashMap<String, CloudSqlInstance> instances = new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<String, ConnectionConfig> namedConfigs =
```

Comment:
Can we achieve this functionality with only one ConcurrentHashMap?

Reply:
This separates the named instances from the unnamed instances. Separate maps is the cleanest way to avoid unintentional conflicts in the key spaces. 

Also, we have been debating whether to change the key type for unnamed instances to include more properties than just the instance name. That implementation would not use the String as the key for unnamed instances.